cli/src/pcluster/templates/cdk_builder_utils.py

```diff
@@ -380,18 +392,46 @@ def _add_node_role(self, node: Union[HeadNode, BaseQueue], name: str):
             role_name=role_name,
         )
 
-    def _add_pcluster_policies_to_role(self, role_ref: str, name: str):
+    def _add_pcluster_policies_to_role(self, role_ref: str, shared_storage_infos: dict, name: str):
         _, policy_name = add_cluster_iam_resource_prefix(
             self._config.cluster_name, self._config, "parallelcluster", iam_type="AWS::IAM::Policy"
         )
+        common_policies = []
+        if self._config.scheduling.scheduler != "awsbatch":
+            efs_with_iam_authorization_arns = self._get_efs_with_iam_authorization_arns(shared_storage_infos)
+            if efs_with_iam_authorization_arns:
+                common_policies.append(
+                    iam.PolicyStatement(
+                        sid="Efs",
+                        actions=[
+                            "elasticfilesystem:ClientMount",
+                            "elasticfilesystem:ClientRootAccess",
+                            "elasticfilesystem:ClientWrite",
+                        ],
+                        effect=iam.Effect.ALLOW,
+                        resources=efs_with_iam_authorization_arns,
+                    ),
+                )
```

Comment:
What about add this logic in the method _build_policy?

Reply:
Adding this logic in _build_policy requires duplication for head and compute nodes.
Head node https://github.com/aws/aws-parallelcluster/blob/develop/cli/src/pcluster/templates/cdk_builder_utils.py#L488
Compute nodes https://github.com/aws/aws-parallelcluster/blob/develop/cli/src/pcluster/templates/cdk_builder_utils.py#L756

So the current place should be the right place to put common policies